src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2086,103 +2311,59 @@ private bool UseUpdatedEscapeRulesForInvocation(Symbol symbol)
             uint scopeOfTheContainingExpression,
             BindingDiagnosticBag diagnostics)
         {
-            // https://github.com/dotnet/csharplang/blob/main/proposals/low-level-struct-improvements.md#rules-method-invocation
-            //
-            // For any method invocation `e.M(a1, a2, ... aN)`
-            // 1. Calculate the *safe-to-escape* of the method return.
-            //     - Ignore the *ref-safe-to-escape* of arguments to `in / ref / out` parameters of `ref struct` types. The corresponding parameters *ref-safe-to-escape* are at most *return only* and hence cannot be returned via `ref` or `out` parameters.
-            //     - Assume the method has a `ref struct` return type.
-            // 2. All `ref` or `out` arguments of `ref struct` types must be assignable by a value with that *safe-to-escape*.
-            //    This applies even when the `ref` argument matches a `scoped ref` parameter.
-
-            if (!symbol.RequiresInstanceReceiver())
-            {
-                // ignore receiver when symbol is static
-                receiverOpt = null;
-            }
```

Comment:
Is this check not needed?

Reply:
Every single caller did this check so I just moved it to the implementation.